haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -56,6 +56,7 @@ def __init__(
         tokenizer_model_folder: Optional[Union[str, Path]] = None,
         language: str = "en",
         id_hash_keys: Optional[List[str]] = None,
+        add_page_number: bool = False,
```

Comment:
Page counting is determined by `"\f"` chars which is added by `PDFToTextConverter`, `ParsrConverter` and `AzureConverter` in between pages. I set the default value here to `False` because the `PreProcessor` can be used with Documents from any origin, not only PDFs. For example, when using the `TextProcessor` on txt-files, we don't have any pages but a single stream of text. If we would set `add_page_number` to `True` on Documents coming from the `TextProcessor`, all the resulting Documents would have the same metada field `"page"` with a value of `1`.

Reply:
👍 thanks for the explanation!